third_party/tonic/logging/dart_error.h

```diff
@@ -7,13 +7,61 @@
 
 #include "third_party/dart/runtime/include/dart_api.h"
 
+#include "tonic/dart_persistent_value.h"
+
 namespace tonic {
 
 namespace DartError {
+using UnhandledExceptionReporter =
+    std::function<void(const std::string&, const std::string&)>;
+
 extern const char kInvalidArgument[];
+extern DartPersistentValue on_error;
+extern UnhandledExceptionReporter log_unhandled_exception;
 }  // namespace DartError
 
-bool LogIfError(Dart_Handle handle);
+/// Check if a Dart_Handle is an error or exception.
+///
+/// If it is an error or exception, this method will return true.
+///
+/// If it is an unhandled error or exception, first a call will be made to the
+/// Dart code set in |SetErrorHandler|. If that closure returns false, a
+/// fallback is made to the closure in |SetUnhandledExceptionReporter| to log
+/// details of the exception and stack. If that closure throws an exception
+/// itself, the exception and stack are logged via the fallback method, and the
```

Comment:
What is the fallback method?

Reply:
The same fallback method mentioned on line 29 - I'll try to word this more clearly.